src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1790,13 +1616,8 @@ internal Solution WithoutFrozenSourceGeneratedDocuments()
         /// </remarks>
         internal Solution WithCachedSourceGeneratorState(ProjectId projectToUpdate, Project projectWithCachedGeneratorState)
         {
-            var newState = _state.WithCachedSourceGeneratorState(projectToUpdate, projectWithCachedGeneratorState);
-            if (newState == _state)
-            {
-                return this;
-            }
-
-            return new Solution(newState);
+            var newCompilationState = _compilationState.WithCachedSourceGeneratorStateWorker(projectToUpdate, projectWithCachedGeneratorState);
```

Comment:
Why the "Worker" suffix this time?

Reply:
whoops.  an artifact of many refactorings.  renamed.